barlayout.go

```diff
@@ -0,0 +1,145 @@
+package desktop
+
+import (
+	"math"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+)
+
+// Declare conformity with Layout interface
+var _ fyne.Layout = (*BarLayout)(nil)
+
+//BarLayout returns a layout used for zooming linear groups of icons
+type BarLayout struct {
+	MouseInside   bool
+	MousePosition fyne.Position
+}
+
+func isHorizontalSpacer(obj fyne.CanvasObject) bool {
+	if spacer, ok := obj.(layout.SpacerObject); ok {
+		return spacer.ExpandHorizontal()
+	}
+
+	return false
+}
+
+//SetPointerInside tells the BarLayout that the mouse is inside of the Layout.
+func (bl *BarLayout) SetPointerInside(inside bool) {
+	bl.MouseInside = inside
+}
+
+//SetPointerPosition tells the BarLayout that the mouse position has been updated.
+func (bl *BarLayout) SetPointerPosition(position fyne.Position) {
+	bl.MousePosition = position
+}
+
+func (bl *BarLayout) isSpacer(obj fyne.CanvasObject) bool {
+	// invisible spacers don't impact layout
+	if !obj.Visible() {
+		return false
+	}
+
+	return isHorizontalSpacer(obj)
+}
+
+// Layout is called to pack all child objects into a specified size.
+// For a VBoxLayout this will pack objects into a single column where each item
+// is full width but the height is the minimum required.
+// Any spacers added will pad the view, sharing the space if there are two or more.
+func (bl *BarLayout) Layout(objects []fyne.CanvasObject, size fyne.Size) {
+	spacers := make([]fyne.CanvasObject, 0)
```

Comment:
We should not have to manage spacers at all - this is centered in the available space so we can code implicit spacers left and right without needing to include objects to represent them on a list?